source/adios2/ADIOS.h

```diff
@@ -97,13 +97,34 @@ class ADIOS
      * @param dimensions
      * @param selections
      * @param offsets
+     * @constantShape true if dimensions, offsets and local sizes don't change
+     * over time
      * @return reference to Variable object
      */
     template <class T>
-    Variable<T> &DefineVariable(const std::string &name,
-                                const Dims localDimensions = Dims{1},
-                                const Dims globalDimensions = Dims{},
-                                const Dims offsets = Dims{});
+    Variable<T> &DefineVariable(const std::string &name, const Dims shape = {},
+                                const Dims start = {}, const Dims count = {},
+                                bool constantShape = false);
+
+    template <class T>
+    Variable<T> &DefineLocalValue(const std::string &name)
+    {
+        return DefineVariable<T>(name, {LocalValueDim}, {}, {}, false);
+    }
+
+    template <class T>
+    Variable<T> &DefineLocalArray(const std::string &name, bool isJoined,
+                                  Dims shape = {})
+    {
+        return DefineVariable<T>(name, shape, {}, {}, false);
+    }
+
+    template <class T>
+    Variable<T> &DefineArray(const std::string &name, Dims shape = {},
```

Comment:
Array + Dims is confusing, I will keep it as Variable since we only deal with 1D arrays.  As a first-time user I'd think I can pass a double**  with a shape = {Nx, Ny}; since we "DefineArray"